### PR TITLE
Upstream error checking

### PR DESCRIPTION
**Pull request by bcorwin**
Update on dashboard needs testing...do we still return back to the
dashboard page?

### CODE REVIEW

File: monitor/views.py [hunk 1]
```diff
@@ -487,47 +487,8 @@ def dashboard(request):
...
-            if Reading.objects.filter(beer=active_beer).order_by("-instant_actual")[:1].get() != old_reading: break
-            sleep(.1)
-    print(command_status[0])
-    return HttpResponseRedirect(reverse('dashboard'))
-def gen_dashboard(readings):
     active_config = getActiveConfig()
-    cur_reading = readings[:1].get()
+    cur_reading = getReadings(getActiveBeer()).order_by("-instant_actual")[:1].get()
     
     active_beer = getActiveBeer()
```
Comment (cutreth) [inline monitor/views.py]: Why not bump this up a few lines and get rid of the duplicate getActiveBeer call in getReadings?

File: monitor/views.py [hunk 2]
```diff
@@ -487,47 +487,8 @@ def dashboard(request):
...
-        if command_status[0] == "Success": break
-        sleep(.1)
-    if command_status[0] == "Success":
-        for i in range(49):
-            if Reading.objects.filter(beer=active_beer).order_by("-instant_actual")[:1].get() != old_reading: break
-            sleep(.1)
-    print(command_status[0])
-    return HttpResponseRedirect(reverse('dashboard'))
-def gen_dashboard(readings):
     active_config = getActiveConfig()
```
Comment (cutreth) [inline monitor/views.py]: Do you ever use active_config? If not, we can delete this line.

File: monitor/views.py [hunk 3]
```diff
@@ -592,3 +585,17 @@ def annotationchart(request, cur_beer = None):
         "beer_date": active_beer.brew_date,
     }
     return render_to_response('annotationchart.html', data)
+def data_chk(request, page_name, cur_beer = None):
+    if cur_beer is None: active_beer = getActiveBeer()
+    else: active_beer = Beer.objects.get(pk=cur_beer)
+    
+    readings = getReadings(active_beer).order_by("-instant_actual")
```
Comment (cutreth) [inline monitor/views.py]: Do we need to return all readings? Can we just grab the first or last one? If we at least have a single reading, we're OK to proceed.

File: monitor/views.py [hunk 4]
```diff
@@ -592,3 +585,17 @@ def annotationchart(request, cur_beer = None):
...
+    
+    if(readings.count() == 0): out = gen_unableToLoad(page_name)
+    else:
+        if page_name.upper() == "DASHBOARD": out = dashboard(readings)
```
Comment (cutreth) [inline monitor/views.py]: After the above change, we choose to not define "readings" in this function. I'd prefer to be consistent with how we hit pages here, so instead we'd just pass on the "request" parameter to the dashboard view like we do for the graphs.

File: monitor/views.py [hunk 5]
```diff
@@ -592,3 +585,17 @@ def annotationchart(request, cur_beer = None):
...
+        elif page_name.upper() == "GRAPH": out = graph(request, cur_beer)
+        elif page_name.upper() == "ANNOTATIONCHART": out = annotationchart(request, cur_beer)
+        else: out = "404 Page Not Found"
```
Comment (cutreth) [inline monitor/views.py]: I believe there's a 404 function we can call actually.

File: monitor/views.py [hunk 6]
```diff
@@ -592,3 +585,17 @@ def annotationchart(request, cur_beer = None):
...
+    
+    return out
```
Comment (cutreth) [inline monitor/views.py]: I'm at work so I haven't tested - does this actually reroute to the pages correctly?

File: monitor/views.py (hunk 2 above)
Comment (bcorwin) [inline monitor/views.py]: Yes, to get temp bounds.

File: monitor/views.py (hunk 5 above)
Comment (bcorwin) [inline monitor/views.py]: That should never be called with the current structure but I added it as a precaution.

File: monitor/views.py (hunk 6 above)
Comment (bcorwin) [inline monitor/views.py]: It worked for me (for what I could test). The update button the dashboard needs testing (I can't do that locally).

File: monitor/views.py (hunk 2 above)
Comment (cutreth) [inline monitor/views.py]: Ahh. Missed that with the way lines get hidden.

**Comment by bcorwin** [comment]
Getting into the minor stuff...I don't have to resort it either.

**Pull request merged**